main.js

```diff
@@ -106,12 +97,33 @@ function favIdea(){
 
 
 function deleteIdea(){
-  console.log(event.target.parentElement.parentElement.id)
   for (var i = 0; i < ideas.length; i++){
     if(ideas[i].id == event.target.parentElement.parentElement.id){
-      console.log("in deleteIdea if");
+      ideas[i].deleteFromStorage();
       ideas.splice(i, 1);
     }
   }
   displayGrid();
 }
+
+// window.addEventListener('load', loadedTracker);
+//
+// function loadedTracker(){
+//   alert("The page has loaded.")
+// }
+window.addEventListener('load', loadSavedIdeas);
+window.addEventListener('load', displayGrid);
+
+function loadSavedIdeas() {
+  var loadedIdea = '';
+  var retrievedIdea = '';
+  for (var i = 0; i < localStorage.length; i++) {
+    loadedIdea = localStorage.key(i);
+    retrievedIdea = JSON.parse(localStorage.getItem(loadedIdea));
+    var idea = new Idea (retrievedIdea.title, retrievedIdea.body);
+    idea.id = retrievedIdea.id;
+    idea.star = retrievedIdea.star;
```

Comment:
I'm so angry at how much trouble this gave us, but I'm glad we figured it out in the end. This still feels... kludgy and inelegant, to me? But I can't think of a way to clean it up. Maybe it'll come to us as we take a look at the code later.